src/backend_data_structure.cc

```diff
@@ -0,0 +1,27 @@
+#include "backend_data_structure.h"
+
+BackendDataStructure::BackendDataStructure() : key_value_map_() {}
+
+bool BackendDataStructure::Put(const std::string &key,
+                               const std::string &value) {
+  key_value_map_[key] = value;
+  return true;
+}
+
+bool BackendDataStructure::Get(const std::string &key,
+                               std::string *output_value) {
+  auto it = key_value_map_.find(key);
+  if (it != key_value_map_.end()) {
+    if (output_value != nullptr) {
```

Comment:
why do you need this nullptr check?

Reply:
Maybe I should have specified this in the ```.h``` file. This offers a benefit that if the caller put ```nullptr``` in ```output_value```, the ```get``` function just checks whether the key passed in exists or not.

Reply:
I feel like that causes your Get() to have 2 different functionalities. One to check existence of key and one to actually retrieve the value

Reply:
Yes! And this is my habit to avoid some misuse that potentially causes segmentation fault.